module/zfs/zvol.c

```diff
@@ -2086,16 +2085,18 @@ zvol_rename_minors_impl(const char *oldname, const char *newname)
 		} else if (strncmp(zv->zv_name, oldname, oldnamelen) == 0 &&
 		    (zv->zv_name[oldnamelen] == '/' ||
 		    zv->zv_name[oldnamelen] == '@')) {
-			snprintf(name, MAXNAMELEN, "%s%c%s", newname,
+			name = kmem_asprintf("%s%c%s", newname,
```

Comment:
nit: `kmem_asprintf()` using `KM_SLEEP` internally and cannot fail so we can drop the `if (name != NULL)` checks.  What do you think about the following to simplify this.

```diff
diff --git a/module/zfs/zvol.c b/module/zfs/zvol.c
index 8890aaa..21d4467 100644
--- a/module/zfs/zvol.c
+++ b/module/zfs/zvol.c
@@ -2063,14 +2063,12 @@ zvol_rename_minors_impl(const char *oldname, const char 
 {
        zvol_state_t *zv, *zv_next;
        int oldnamelen, newnamelen;
-       char *name;
 
        if (zvol_inhibit_dev)
                return;
 
        oldnamelen = strlen(oldname);
        newnamelen = strlen(newname);
-       name = kmem_alloc(MAXNAMELEN, KM_SLEEP);
 
        mutex_enter(&zvol_state_lock);
 
@@ -2086,16 +2084,15 @@ zvol_rename_minors_impl(const char *oldname, const char 
                } else if (strncmp(zv->zv_name, oldname, oldnamelen) == 0 &&
                    (zv->zv_name[oldnamelen] == '/' ||
                    zv->zv_name[oldnamelen] == '@')) {
-                       snprintf(name, MAXNAMELEN, "%s%c%s", newname,
+                       char *name = kmem_asprintf("%s%c%s", newname,
                            zv->zv_name[oldnamelen],
                            zv->zv_name + oldnamelen + 1);
                        zvol_rename_minor(zv, name);
+                       kmem_free(name, strlen(name + 1));
                }
        }
 
        mutex_exit(&zvol_state_lock);
-
-       kmem_free(name, MAXNAMELEN);
 }
 
 typedef struct zvol_snapdev_cb_arg {
```